samples/gen-go-db-custom-path/db/interface.go

```diff
@@ -104,6 +104,23 @@ type Interface interface {
 	// ScanThingsByNameAndCreatedAt runs a scan on the NameAndCreatedAt index.
 	ScanThingsByNameAndCreatedAt(ctx context.Context, input ScanThingsByNameAndCreatedAtInput, fn func(m *models.Thing, lastThing bool) bool) error
 
+	// SaveThingAllowingBatchWrites saves a ThingAllowingBatchWrites to the database.
+	SaveThingAllowingBatchWrites(ctx context.Context, m models.ThingAllowingBatchWrites) error
+	// SaveArrayOfThingAllowingBatchWrites batch saves all items in ArrayOfThingAllowingBatchWrites to the database.
+	SaveArrayOfThingAllowingBatchWrites(ctx context.Context, m models.ArrayOfThingAllowingBatchWrites) error
```

Comment:
does this require defining an array type? I guess I was expecting not to have to specify an array type and instead have something like

```yaml
x-db:
  AllowBatchWrites: true
```

```go
SaveArrayOfThingAllowingBatchWrites(ctx context.Context, ms []models.ThingAllowingBatchWrites) 
```

Reply:
Unfortunately it does require defining an array. The only reason was because swagger deals with the validation for the models and I couldn't figure out how to get that working without a lot of handrolling. Do you have any suggestions?

Reply:
Can you expand on what you mean by
>  swagger deals with the validation for the models

Reply:
Ah actually looking at it now, I was confusing where the `encode` was coming from within the batch saves itself. Let me take another stab at this where we don't need to define the array!